### PR TITLE
`Lectures`: Fix the order of lecture units when merging pdf files

### PR DESCRIPTION
**Pull request by Strohgelaender**
#### General
- [x] This is a small issue that I tested locally and was confirmed by another developer on a test server.
- [x] I chose a title conforming to the [naming conventions for pull requests](https://artemis-platform.readthedocs.io/en/latest/dev/guidelines/development-process.html#naming-conventions-for-github-pull-requests).
#### Server
- [x] I followed the [coding and design guidelines](https://docs.artemis.ase.in.tum.de/dev/guidelines/server/).
- [x] I added multiple integration tests (Spring) related to the features (with a high test coverage).

### Motivation and Context
When using the merge pdf feature as a student, the created pdf file contained all slides in the order of the upload of their respective units. This is an issue e.g. when last minute changes are necessary, sicnce recreating a unit breaks the order of the merged file.

Testing the merge feature as a editor is also quite hard right now. The merged pdf only contains already public lecture units. This should be changed for tutors / editors, their file should contain all 
### Description
When merging the files, they are now sorted by the order-attribute.

Aditionally, we now use `isAllowedToSeeLectureUnit()`. This method checks the role of the user, allowing tutors and above to create a merged pdf including hidden units. With the current design of the file ressource, we cannot use the usual aproach of calling this method with `null` and loading the user automatically. Instead, we have to parse the username from the token.

### Steps for Testing
Prerequisites:
- 1 Instructor, 1 Student, 1 Tutor

1. Create two new lecture units with pdf attatchments:
2. As a student, click the "merge pdfs" button
3. Change the order of the units as instructor
4. Re-Download the merged pdfs again
5. Ensure that in both downloads the order was used, meaning the second download should have the files fliped.
6. Add another unit with a release date in the future
7. Merge the PDF as student and tutor. The students' pdf should not contain the new unit, but it should be included into the result for the tutor.

#### Code Review
- [x] Review 1
- [x] Review 2
#### Manual Tests
- [x] Test 1
- [x] Test 2

### Test Coverage
| Class/File | Branch | Line |
|------------|-------:|-----:|

### CODE REVIEW

**Review by Marvin-Vitcu — APPROVED** [review]
tested with @Strohgeländer on his local setup. Everything worked as expected locally.

**Review by mallgaier — APPROVED** [review]
Tested on TS3, works as expected.
Excellent job 👍

**Review by max-bergmann — APPROVED** [review]
Tested on TS3, works as expected.

File: src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java [hunk 1]
```diff
@@ -350,13 +350,19 @@ public ResponseEntity<byte[]> getLecturePdfAttachmentsMerged(@PathVariable Long
...
+        // Parse user information from the access token
+        var claims = this.tokenProvider.parseClaims(temporaryAccessToken);
+        String username = claims.getSubject();
+        User user = userRepository.getUserWithGroupsAndAuthorities(username);
+
         Set<AttachmentUnit> lectureAttachments = attachmentUnitRepository.findAllByLectureIdAndAttachmentTypeElseThrow(lectureId, AttachmentType.FILE);
 
         List<String> attachmentLinks = lectureAttachments.stream()
-                .filter(unit -> unit.isVisibleToStudents() && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
+                .filter(unit -> authCheckService.isAllowedToSeeLectureUnit(unit, user) && "pdf".equals(StringUtils.substringAfterLast(unit.getAttachment().getLink(), ".")))
```
Comment (ge65cer) [inline src/main/java/de/tum/in/www1/artemis/web/rest/FileResource.java:361]: If you pass `null` as the user, the method itself will fetch it from the `userRepository`. Sure, you have some overhead from the loop -- on the other side you don't need to inject the repository into this class. What performs better?

**Review by willergeorg — APPROVED** [review]
Code lgtm. 👍

**Review by ge65cer — APPROVED** [review]
Questions clarified via PM, approved!

**Pull request merged**